### PR TITLE
Change args for install_deps in Make file

### PR DESCRIPTION
**Pull request by tonygardella**
## Description
Latest release for the R package "remotes" has [breaking changes](https://github.com/r-lib/remotes/releases/tag/v2.0.0) for the `install_deps` function. The `threads` arg need to be changes Ncpus.

THIS HAS NOT SOLVED THE BUILD LOCALLY. I'm still running into the issues locally with make where it can't install PEcAn packages.
## Motivation and Context
Travis builds are breaking installing all PEcAn packages.
## Review Time Estimate

- [x] Immediately
- [ ] Within one week
- [ ] When possible
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by tonygardella** [comment]
It seems to install the PEcAN packages, but down the line it says they're not there:
```
time Rscript -e "devtools::install('base/remote', Ncpus = 1);"

Skipping 1 packages not available: PEcAn.logger
   checking for file ‘/fs/data3/tonygard/work/pecan/base/remote/DESCRIPTION’ .✔  checking for file ‘/fs/data3/tonygard/work/pecan/base/remote/DESCRIPTION’
─  preparing ‘PEcAn.remote’:
✔  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘PEcAn.remote_1.6.0.tar.gz’
   Warning: invalid uid value replaced by that for user 'nobody'
   Warning: invalid gid value replaced by that for user 'nobody'

Running /usr/lib/R/bin/R CMD INSTALL \
  /tmp/RtmpC2TqiK/PEcAn.remote_1.6.0.tar.gz --install-tests
* installing to library ‘/fs/data3/tonygard/R/library’
* installing *source* package ‘PEcAn.remote’ ...
** R
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (PEcAn.remote)

real    0m5.667s
user    0m4.156s
sys     0m0.384s
echo `date` > .install/base/remote
time Rscript -e "devtools::install_deps('base/utils', Ncpus = 1, dependencies = TRUE);"

Skipping 7 packages not available: PEcAn.data.atmosphere, PEcAn.data.land, PEcAn.DB, PEcAn.emulator, PEcAn.logger, PEcAn.remote, PEcAn.settings

real    0m4.432s
user    0m3.676s
sys     0m0.144s
Rscript -e "devtools::document('"base/utils"')"
Updating PEcAn.utils documentation
Writing NAMESPACE
Loading PEcAn.utils
Writing NAMESPACE
echo `date` > .doc/base/utils
time Rscript -e "devtools::install('base/utils', Ncpus = 1);"

Skipping 2 packages not available: PEcAn.logger, PEcAn.remote
   checking for file ‘/fs/data3/tonygard/work/pecan/base/utils/DESCRIPTION’ ..✔  checking for file ‘/fs/data3/tonygard/work/pecan/base/utils/DESCRIPTION’
─  preparing ‘PEcAn.utils’:
✔  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  looking to see if a ‘data/datalist’ file should be added
─  building ‘PEcAn.utils_1.6.0.tar.gz’
   Warning: invalid uid value replaced by that for user 'nobody'
   Warning: invalid gid value replaced by that for user 'nobody'

Running /usr/lib/R/bin/R CMD INSTALL /tmp/RtmpgVKHJp/PEcAn.utils_1.6.0.tar.gz \
  --install-tests
* installing to library ‘/fs/data3/tonygard/R/library’
* installing *source* package ‘PEcAn.utils’ ...
** R
** data
*** moving datasets to lazyload DB
** inst
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (PEcAn.utils)

real    0m7.489s
user    0m6.044s
sys     0m0.452s
echo `date` > .install/base/utils
time Rscript -e "devtools::install_deps('base/db', Ncpus = 1, dependencies = TRUE);"

Skipping 3 packages not available: PEcAn.logger, PEcAn.remote, PEcAn.utils
```

File: Makefile [hunk 1]
```diff
@@ -129,7 +129,7 @@ clean:
 	+ time Rscript -e "if(!requireNamespace('mockery', quietly = TRUE)) install.packages('mockery', repos = 'http://cran.rstudio.com', Ncpus = ${NCPUS})"
 	echo `date` > $@
 
-depends_R_pkg = time Rscript -e "devtools::install_deps('$(strip $(1))', threads = ${NCPUS}, dependencies = TRUE);"
+depends_R_pkg = time Rscript -e "devtools::install_deps('$(strip $(1))', Ncpus = ${NCPUS}, dependencies = TRUE);"
```
Comment (infotroph) [inline Makefile]: Just to confirm: Are we still OK to call it from devtools, or or do we also need to switch the spelling to `remotes::install_deps`?

Comment (tonygardella) [reply Makefile]: OOhhhh yeah didn't even check that!

Comment (tonygardella) [reply Makefile]: hmmm... that doesn't do it

**Comment by tonygardella** [comment]
So this is how Make starts, it says it can load logger but then it says it's not available:
```
mkdir -p .doc/base
mkdir -p .install
time Rscript -e "if(!requireNamespace('devtools', quietly = TRUE)) install.packages('devtools', repos = 'http://cran.rstudio.com', Ncpus = 1)"

real    0m1.301s
user    0m1.020s
sys     0m0.068s
echo `date` > .install/devtools
time Rscript -e "if(!requireNamespace('remotes', quietly = TRUE)) install.packages('remotes', repos = 'http://cran.rstudio.com', Ncpus = 1)"

real    0m0.110s
user    0m0.080s
sys     0m0.020s
echo `date` > .install/remotes
time Rscript -e "if(!requireNamespace('roxygen2', quietly = TRUE)) install.packages('roxygen2', repos = 'http://cran.rstudio.com', Ncpus = 1)"

real    0m0.427s
user    0m0.312s
sys     0m0.044s
echo `date` > .install/roxygen2
time Rscript -e "if(!requireNamespace('testthat', quietly = TRUE)) install.packages('testthat', repos = 'http://cran.rstudio.com', Ncpus = 1)"

real    0m0.224s
user    0m0.180s
sys     0m0.020s
echo `date` > .install/testthat
time Rscript -e "remotes::install_deps('base/logger', Ncpus = 1, dependencies = TRUE);"


real    0m3.028s
user    0m2.292s
sys     0m0.108s
Rscript -e "devtools::document('"base/logger"')"
Updating PEcAn.logger documentation
Writing NAMESPACE
Loading PEcAn.logger
Writing NAMESPACE
echo `date` > .doc/base/logger
mkdir -p .install/base
time Rscript -e "devtools::install('base/logger', Ncpus = 1);"
✔  checking for file ‘/fs/data3/tonygard/work/pecan/base/logger/DESCRIPTION’ ...
─  preparing ‘PEcAn.logger’:
✔  checking DESCRIPTION meta-information
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘PEcAn.logger_1.6.0.tar.gz’
   Warning: invalid uid value replaced by that for user 'nobody'
   Warning: invalid gid value replaced by that for user 'nobody'

Running /usr/lib/R/bin/R CMD INSTALL \
  /tmp/RtmpfQ7ht6/PEcAn.logger_1.6.0.tar.gz --install-tests
* installing to library ‘/fs/data3/tonygard/R/library’
* installing *source* package ‘PEcAn.logger’ ...
** R
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (PEcAn.logger)

real    0m4.782s
user    0m3.716s
sys     0m0.392s
echo `date` > .install/base/logger
time Rscript -e "remotes::install_deps('base/remote', Ncpus = 1, dependencies = TRUE);"

Skipping 1 packages not available: PEcAn.logger
```

**Comment by tonygardella** [comment]
Maybe the warnings are an actual issue?: 
```
─  building ‘PEcAn.logger_1.6.0.tar.gz’
   Warning: invalid uid value replaced by that for user 'nobody'
   Warning: invalid gid value replaced by that for user 'nobody'
```

**Comment by infotroph** [comment]
🤷‍♂️ Did the install step always build a tarball, or is that new too?

**Comment by tonygardella** [comment]
@infotroph ah yep they switched from zip to tar.

**Comment by infotroph** [comment]
@tonygardella How far through a local build can you get so far?

**Comment by tonygardella** [comment]
@infotroph Rechecking now. It gets pretty far then says 22 packages aren't available, which is farther than TRAVIS right now. I'll yet you know when it's done. I'm looking into what the change to tar might break/ what new args we need.

**Comment by tonygardella** [comment]
Also, no need to hassle yourself too much. Hamze is helping me now and hopefully others can jump in too.

**Comment by tonygardella** [comment]
@infotroph It gets to building PEcAn.all locally, then says:
```
time Rscript -e "remotes::install_deps('base/all', Ncpus = 1, dependencies = TRUE);"

Skipping 22 packages not available: PEcAn.allometry, PEcAn.assim.batch, PEcAn.benchmark, PEcAn.biocro, PEcAn.DALEC, PEcAn.data.atmosphere, PEcAn.data.land, PEcAn.data.remote, PEcAn.DB, PEcAn.ed, PEcAn.emulator, PEcAn.LINKAGES, PEcAn.logger, PEcAn.MA, PEcAn.photosythesis, PEcAn.priors, PEcAn.remote, PEcAn.settings, PEcAn.sipnet, PEcAn.uncertainty, PEcAn.utils, PEcAn.workflow
Installing 4 packages: PEcAn.biocro, PEcAn.ed, PEcAn.photosythesis, PEcAn.sipnet
Installing packages into ‘/fs/data3/tonygard/R/library’
(as ‘lib’ is unspecified)
Error: (converted from warning) packages ‘PEcAn.biocro’, ‘PEcAn.ed’, ‘PEcAn.photosythesis’, ‘PEcAn.sipnet’ are not available (for R version 3.4.4)
Execution halted

real    0m3.964s
user    0m3.264s
sys     0m0.120s
Makefile:146: recipe for target '.doc/base/all' failed
make: *** [.doc/base/all] Error 1
```

**Comment by infotroph** [comment]
So basically it looks as if install_deps used to only warn about dependencies it doesn't know how to install, but now raises the warning to an error unless they're already available -- hence the local build not dying until it tried the nonexistant packages. Earlier failure on Travis could mean maybe it's not looking in the right R library?

**Comment by tonygardella** [comment]
@infotroph Not sure if it's just TRAVIS not findgin the right R library. I think there are some changes to defining dependencies because as soon as I set `dependencies= FALSE` the error goes away.

**Comment by tonygardella** [comment]
Ah I think it's the `upgrade` argument. That's a recent change I need to read up on.

**Comment by tonygardella** [comment]
For documenting purposes When upgrade set to FALSE you get this error:
```
✔  checking for file ‘/home/travis/build/PecanProject/pecan/base/remote/DESCRIPTION’
─  preparing ‘PEcAn.remote’:
✔  checking DESCRIPTION meta-information
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘PEcAn.remote_1.6.0.tar.gz’
   
Running /home/travis/R-bin/lib/R/bin/R CMD INSTALL \
  /tmp/Rtmpx09InY/PEcAn.remote_1.6.0.tar.gz --install-tests 
* installing to library ‘/home/travis/R/Library’
* installing *source* package ‘PEcAn.remote’ ...
** R
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (PEcAn.remote)
3.36user 0.22system 0:03.40elapsed 105%CPU (0avgtext+0avgdata 98800maxresident)k
0inputs+2256outputs (0major+104271minor)pagefaults 0swaps
echo `date` > .install/base/remote
time Rscript -e "remotes::install_deps('base/utils', Ncpus = 2, dependencies = TRUE, upgrade = FALSE);"
2.08user 0.05system 0:02.23elapsed 95%CPU (0avgtext+0avgdata 67740maxresident)k
0inputs+672outputs (0major+21936minor)pagefaults 0swaps
Rscript -e "devtools::document('"base/utils"')"
Updating PEcAn.utils documentation
Writing NAMESPACE
Loading PEcAn.utils
Error: Dependency package(s) 'abind','data.table','getPass','ncdf4','PeriodicTable','randtoolbox','rjags','udunits2' not available.
In addition: Warning messages:
1: In (function (dep_name, dep_ver = "*")  :
  Dependency package 'abind' not available.
2: In (function (dep_name, dep_ver = "*")  :
  Dependency package 'data.table' not available.
3: In (function (dep_name, dep_ver = "*")  :
  Dependency package 'getPass' not available.
4: In (function (dep_name, dep_ver = "*")  :
  Dependency package 'ncdf4' not available.
5: In (function (dep_name, dep_ver = "*")  :
  Dependency package 'PeriodicTable' not available.
6: In (function (dep_name, dep_ver = "*")  :
  Dependency package 'randtoolbox' not available.
7: In (function (dep_name, dep_ver = "*")  :
  Dependency package 'rjags' not available.
8: In (function (dep_name, dep_ver = "*")  :
  Dependency package 'udunits2' not available.
Execution halted
make: *** [.doc/base/utils] Error 1
```

**Pull request closed without merging**